Src/FluentAssertions/Common/TaskTimer.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace FluentAssertions.Common
+{
+    /// <summary>
+    /// Default implementation for <see cref="ITimer"/> for production use.
+    /// </summary>
+    internal class TaskTimer : ITimer
+    {
+        public Task DelayAsync(TimeSpan delay)
+        {
+            return Task.Delay(delay);
+        }
+
+        public bool Wait(Task task, TimeSpan timeout)
+        {
+            return task.Wait(timeout);
```

Comment:
This can cause deadlocks (which seem to happen more often on AppVeyor). Consider using a construct like https://github.com/ChillBDD/Chill/blob/master/Src/Chill/Common/TaskExtensions.cs#L16 or use a `Task.Run`.

Reply:
I have introduced solution provided by #1027.